sdcm/nemesis.py

```diff
@@ -588,33 +590,70 @@ def disrupt_nodetool_refresh(self, big_sstable=True, skip_download=False):
             sstable_url = 'https://s3.amazonaws.com/scylla-qa-team/refresh_nemesis/keyspace1.standard1.100M.tar.gz'
             sstable_file = '/tmp/keyspace1.standard1.100M.tar.gz'
             sstable_md5 = '9c5dd19cfc78052323995198b0817270'
+            keys_num = 501000
         else:
             sstable_url = 'https://s3.amazonaws.com/scylla-qa-team/refresh_nemesis/keyspace1.standard1.tar.gz'
             sstable_file = "/tmp/keyspace1.standard1.tar.gz"
             sstable_md5 = 'c033a3649a1aec3ba9b81c446c6eecfd'
-        if not skip_download:
-            key_store = KeyStore()
-            creds = key_store.get_scylladb_upload_credentials()
-            remote_get_file(self.target_node.remoter, sstable_url, sstable_file,
-                            hash_expected=sstable_md5, retries=2,
-                            user_agent=creds['user_agent'])
+            keys_num = 1000
+
+        query_cmd = "SELECT * FROM keyspace1.standard1 LIMIT 1"
+        result = self.target_node.run_cqlsh(query_cmd)
+        cols = re.findall("(\| C\d+)", result.stdout)
+        if len(cols) == 1:
+            # Use special schema (one column) for refresh before https://github.com/scylladb/scylla/issues/6617 is fixed
+            if big_sstable:
+                sstable_url = 'https://s3.amazonaws.com/scylla-qa-team/refresh_nemesis_c0/keyspace1.standard1.100M.tar.gz'
+                sstable_file = '/tmp/keyspace1.standard1.100M.tar.gz'
+                sstable_md5 = 'e4f6addc2db8e9af3a906953288ef676'
+                keys_num = 1001000
+            else:
+                sstable_url = 'https://s3.amazonaws.com/scylla-qa-team/refresh_nemesis_c0/keyspace1.standard1.tar.gz'
+                sstable_file = "/tmp/keyspace1.standard1.tar.gz"
+                sstable_md5 = 'c4aee10691fa6343a786f52663e7f758'
+                keys_num = 1000
```

Comment:
I think that this new code including the above old code should be:

col_num = check_number_of_columns.
if col_num = 1:
     if big_sstable:
        url = x
     else:
        url = y
elsif col_num >=5:
    if big_sstable:
         url = z
    else:
         url = w
else:
     raise "Schema doesn't match the snapshot, not uploading".

Reply:
In my latest test result, snapshot (col=5) can be loaded to table (col=20).
they rest columns will be filled to 'null'.

Our SELECT failed for DELETE in the old testing, I wrongly treat it as a scylla issue.

So it's safe to use above snapshot for rest columns.

Current column types:
- 1
- 5
- 8
- 16
- 20
- 512

We don't have columns  (1 < cols < 5), so the issue #6617 won't be triggered

Reply:
I will change the code to:

```
col_num = check_number_of_columns.
if col_num >= 5:
     if big_sstable:
        url = x
     else:
        url = y
else:
    if big_sstable:
         url = z
    else:
         url = w

```

Reply:
As we discussed in hangout, I will change the code as you suggested in https://github.com/scylladb/scylla-cluster-tests/pull/2266#discussion_r462181146

Reply:
Fixed.